### PR TITLE
[LTC] Sync gen_lazy_tensor.py from the staging branch

### PR DESCRIPTION
**Pull request by alanwaketan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #70385

Summary:
This commit sync gen_lazy_tensor.py from the lazy_tensor_staging branch
to the master.

Test Plan:
CI in the lazy_tensor_staging branch.

Differential Revision: [D33306232](https://our.internmc.facebook.com/intern/diff/D33306232)

### CODE REVIEW

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33306232).

File: tools/codegen/gen_lazy_tensor.py [hunk 1]
```diff
@@ -180,26 +181,6 @@ def gen_key(func: FunctionSchema) -> Tuple[str, str]:
         )),
     })
     # Generate headers for shape/dtype funcs for non-meta kernels
-    fm.write_with_template(f'{backend_key}ShapeInference.h', 'ShapeInference.h', lambda: {
```
Comment (bdhirsh) [inline tools/codegen/gen_lazy_tensor.py:183]: Was this duplicated further down - that's why we're deleting it?

Comment (alanwaketan) [reply tools/codegen/gen_lazy_tensor.py:183]: Yes, you are right.

File: tools/codegen/gen_lazy_tensor.py [hunk 2]
```diff
@@ -180,26 +181,6 @@ def gen_key(func: FunctionSchema) -> Tuple[str, str]:
...
-            "vector",
-        ]],
-        'lazy_ir_inc': [],
-        'DispatchKey': backend_key,
-        'dispatch_namespace': backend_key.lower(),
-        'func_declarations': list(concat_map_codegen(
-            dest.GenLazyShapeInferenceDefinition(backend_indices[backend_key],
-                                                 tensor_class),
-            grouped_native_functions,
-            codegenInplaceVariant=True
```
Comment (wconstab) [inline tools/codegen/gen_lazy_tensor.py:199]: but it's not quite a duplicate: the remaining copy does not have codegenInplaceVariant=True - are you sure which version is correct to keep?

Comment (alanwaketan) [reply tools/codegen/gen_lazy_tensor.py:199]: The one on the branch does't have the `codegenInplaceVariant=True` which I assume is the latest one: https://github.com/pytorch/pytorch/blob/lazy_tensor_staging/tools/codegen/gen_lazy_tensor.py.

Comment (wconstab) [reply tools/codegen/gen_lazy_tensor.py:199]: ok. i double checked, i think this is right.

**Review by wconstab — APPROVED** [review]

**Comment by alanwaketan** [comment]
Thanks, Will.

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33306232).

**Pull request closed without merging**